src/categorical_algebra/CSets.jl

```diff
@@ -538,30 +538,74 @@ For each hom in the schema, e.g. h: m → n, the following square must commute:
 Xₕ ↓  ✓  ↓ Yₕ
   Xₙ --> Yₙ
      αₙ 
-
-If `return_failures` is true, return a list of violations, with elements such as: 
-  (h, index i in Xₘ, Yₕ(αₘ(i)), αₙ(Xₕ(i)))
-
-This is type-unstable, so it should not be used in performance-sensitive code.
+You're allowed to run this on a named tuple partly specifying an ACSetTransformation,
+though at this time the domain and codomain must be fully specified ACSets.
+"""
+function is_natural(α::ACSetTransformation) 
```

Comment:
This function should dispatch on `TightACSetTransformatio` vs `LooseACSetTransformation` rather than doing a type check using `isa` in the body.